scipy/special/wright_bessel.pxd

```diff
@@ -0,0 +1,685 @@
+@cython.cdivision(True)
+cdef inline double _wb_series(double a, double b, double x,
+    unsigned int nstart, unsigned int nstop) nogil:
+    cdef:
+        unsigned int k, k_max
+        double xk_k, res
+
+    xk_k = x**nstart * Gamma(nstart + 1)  # x^k/k!
+    res = xk_k * rgamma(nstart*a + b)
+    # term k=nstart+1 , +2 +3, ...
+    if nstop > nstart:
+        # series expansion until term k such that a*k+b <= rgamma_zero
+        k_max = int(floor((rgamma_zero - b) / a))
+        if nstop > k_max:
+            nstop = k_max
+        for k in range(nstart + 1, nstop):
+            xk_k *= x/k
+            res += xk_k * rgamma(a*k + b)
+
+    return res
+
+
+# 2. Taylor series expansion in x=0, for large a.
+#    Phi(a, b, x) = sum_k x^k / k! / Gamma(a*k+b)
+#
+#    Use Stirling formula to find k=k_max, the maximum term.
+#    Then use n terms of Taylor series around k_max.
+@cython.cdivision(True)
+cdef inline double _wb_large_a(double a, double b, double x,
+    unsigned int n) nogil:
+    cdef:
+        unsigned int k, k_max
+        int n_start
+        double lnx, res
+
+    k_max = int(pow(pow(a, -a) * x, 1./(1 + a)))
+    n_start = k_max - n//2
+    if n_start < 0:
+        n_start = 0
+
+    res = 0
+    lnx = log(x)
+    for k in range(n_start, n_start + n):
+        res += exp(k * lnx - lgam(k + 1) - lgam(a*k + b))
+
+    return res
+
+
+# 3. Taylor series in a=0 up to order 5, for tiny a and not too large x
+# Phi(a, b, x) = exp(x)/Gamma(b)
+#               * (1 - a*x*Psi(b) + a^2/2*x*(1+x)(Psi(b)^2 - Psi'(b)) + O(a^3))
+# where Psi is the digamma function.
+# Call: python _precompute/wright_bessel.py 1
+#
+# For small b, cancellation of poles of digamma(b)/Gamma(b) and polygamma needs
+# to be carried out => series expansion in a and b around 0 to order 5.
+# Call: python _precompute/wright_bessel.py 2
+@cython.cdivision(True)
+cdef inline double _wb_small_a(double a, double b, double x, int order) nogil:
+    cdef:
+        double dg, dg1, dg2, dg3, res
+        double A[6]  # powers of a^k/k!
+        double B[5]  # powers of b^k/k!
+        double C[5]  # coefficients of a^k1 * b^k2
+        int k
+
+    if b <= 1e-3:
+        # Series expansion of both a and b up to order 5:
+        C[0] = 1
+        C[1] = 2*M_EG
+        C[2] = 3*M_EG**2 - M_PI**2/2
+        C[3] = 4*M_EG**3 - 2*M_EG*M_PI**2 + 8*M_Z3
+        C[4] = 5*M_EG**4 - 5*M_EG**2*M_PI**2 + 40*M_EG*M_Z3 + M_PI**4/12
+        A[0] = 1.
+        B[0] = 1.
+        for k in range(1, 5):
+            A[k] = a/k * A[k-1]
+            B[k] = b/k * B[k-1]
+        A[5] = a/5*A[4]
+        res = rgamma(b)
+        res += a*x * (C[0] + C[1]*b + C[2]*B[2] + C[3]*B[3] + C[4]*B[4])
+        res += A[2]*x*(1+x)*(C[1]   + C[2]*b    + C[3]*B[2] + C[4]*B[3])
+        res += A[3]*x*(x**2+3*x+1) * (C[2]      + C[3]*b    + C[4]*B[2])
+        res += A[4]*x*(x**3+6*x**2+7*x+1) *      (C[3]      + C[4]*b)
+        res += A[5]*x*(x**4+10*x**3+25*x**2+15*x+1) *         C[4]
```

Comment:
A lot of powers of `x` are getting recomputed here. Is there some recurrence relation for the polynomials in `x`? (Like probably right?)

Reply:
Not that I know of or is evident to me. I put precomputation of powers up to x**4.